### PR TITLE
[CI][nightly-test] test death of threaded actors

### PR DESCRIPTION
**Pull request by scv119**
Ensure we test threaded actors in our nightly tests
- Running test: https://beta.anyscale.com/o/anyscale-internal/projects/prj_2xR6uT6t7jJuu1aCwWMsle/clusters/ses_vDiAWnSWnyUjKY4jGMGFNivV?command-history-section=command_history

### CODE REVIEW

**Comment by scv119** [comment]
oh well, it fails badly

**Comment by simon-mo** [comment]
while you are at it, can we also test [async actor](https://docs.ray.io/en/latest/async_api.html#asyncio-for-actors)? so it can catch https://github.com/ray-project/ray/issues/18533

**Comment by scv119** [comment]
yeah, any scenario you have on your mind? otherwise I just create some random async actors.

**Comment by simon-mo** [comment]
Random one sounds fine.

File: release/nightly_tests/nightly_tests.yaml [hunk 1]
```diff
@@ -221,7 +221,7 @@
 
   run:
     timeout: 7200
-    script: python stress_tests/test_dead_actors.py
+    script: python stress_tests/test_dead_actors.py --num-threads=5
```
Comment (rkooo567) [inline release/nightly_tests/nightly_tests.yaml:224]: Why don't we actually keep the original test (as it is the most common use case) and have a separate tests for threaded actors?

**Comment by iycheng** [comment]
It looks like will fix this as well https://github.com/ray-project/ray/issues/18995

**Comment by rkooo567** [comment]
Is this in progress now?

**Comment by rkooo567** [comment]
```
ray::Parent.ping() (pid=177, ip=172.31.30.195, repr=<test_dead_actors.Parent object at 0x7f15cd9d1110>)
  File "stress_tests/test_dead_actors.py", line 53, in ping
  File "stress_tests/test_dead_actors.py", line 39, in __init__
  File "stress_tests/test_dead_actors.py", line 39, in <listcomp>
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/actor.py", line 525, in remote
    runtime_env=new_runtime_env)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/actor.py", line 616, in _remote
    raise ValueError("max_concurrency must be >= 1")
ValueError: max_concurrency must be >= 1
```
Maybe this is the root cause? When I tried it passed :(... I wonder if it is related to this

File: release/nightly_tests/stress_tests/test_dead_actors.py [hunk 2]
```diff
@@ -85,7 +91,7 @@ def parse_script_args():
                 ray.cluster_resources())
 
     parents = [
-        Parent.remote(num_children, death_probability)
+        Parent.remote(num_children, death_probability, num_threads)
```
Comment (rkooo567) [inline release/nightly_tests/stress_tests/test_dead_actors.py:94]: I think it should be

`Parent.options(max_concurrency=num_threads).remote(...)`

**Comment by scv119** [comment]
no longer needed, @rkooo567 already merged similar PRs into master!

**Pull request closed without merging**